ts/Data/DataConverter.ts

```diff
@@ -0,0 +1,646 @@
+            timestamp = this.parseDate(this.asString(value));
+        }
+
+        return new Date(timestamp);
+    }
+
+    /**
+     * Converts a value to a number.
+     *
+     * @param {DataConverter.Type} value
+     * Value to convert.
+     *
+     * @return {number}
+     * Converted value as a number.
+     */
+    public asNumber(value: DataConverter.Type): number {
+        if (typeof value === 'number') {
+            return value;
+        }
+        if (typeof value === 'boolean') {
+            return value ? 1 : 0;
+        }
+        if (typeof value === 'string') {
+            if (value.indexOf(' ') > -1) {
+                value = value.replace(/\s+/g, '');
+            }
+            if (this.decimalRegex) {
+                value = value.replace(this.decimalRegex, '$1.$2');
+            }
+            return parseFloat(value);
+        }
+        if (value instanceof Date) {
+            return value.getDate();
+        }
+        if (value) {
+            return value.getRowCount();
+        }
+
+        return NaN;
+    }
+
+    /**
+     * Converts a value to a string.
+     *
+     * @param {DataConverter.Type} value
+     * Value to convert.
+     *
+     * @return {string}
+     * Converted value as a string.
+     */
+    public asString(value: DataConverter.Type): string {
+        return `${value}`;
+    }
+
+    /**
+     * Trim a string from whitespaces.
+     *
+     * @param {string} str
+     * String to trim.
+     *
+     * @param {boolean} [inside=false]
+     * Remove all spaces between numbers.
+     *
+     * @return {string}
+     * Trimed string
+     */
+    public trim(
+        str: string,
+        inside?: boolean
+    ): string {
+        const converter = this;
+
+        if (typeof str === 'string') {
+            str = str.replace(/^\s+|\s+$/g, '');
+
+            // Clear white space insdie the string, like thousands separators
+            if (inside && /^[0-9\s]+$/.test(str)) {
+                str = str.replace(/\s/g, '');
+            }
+
+            if (converter.decimalRegex) {
+                str = str.replace(converter.decimalRegex, '$1.$2');
+            }
+        }
+
+        return str;
+    }
+
+    /**
+     * Guesses the potential type of a string value
+     * (for parsing CSV etc)
+     *
+     * @param {string} value
+     * The string to examine
+     * @return {'number'|'string'|'Date'}
+     * `string`, `Date` or `number`
+     */
+    public guessType(
+        value: string
+    ): ('number'|'string'|'Date') {
+        const converter = this,
+            trimVal = converter.trim(value),
+            trimInsideVal = converter.trim(value, true),
+            floatVal = parseFloat(trimInsideVal);
+
+        let result: ('string' | 'Date' | 'number') = 'string',
+            dateVal;
+
+        // is numeric
+        if (+trimInsideVal === floatVal) {
+
+            // If the number is greater than milliseconds in a year, assume
+            // datetime.
```

Comment:
There's some regression, it fails to guess Date here: https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/data/delimiters/

Reply:
Is it fixed now? When I do a local test on the branch, it looks the same with your fiddle. I need more information about the failure to continue.

Reply:
<img width="1172" alt="Skjermbilde 2021-05-31 kl  12 41 45" src="https://user-images.githubusercontent.com/40056287/120181619-9c71d000-c20d-11eb-9aad-3570fd3cf4a7.png">

Reply:
Seems to still be broken for me: https://jsfiddle.net/goransle/9437yp8g/ (also note that the CSV in the sample is slightly wrong on the first line 😬 )

Reply:
Now I see. Chrome breaks, Firefox and Safari not.

Reply:
Please confirm, that it works in Chrome now.

Reply:
It works! 🙌